sklearn/linear_model/_omp.py

```diff
@@ -110,13 +110,13 @@ def _cholesky_omp(X, y, n_nonzero_coefs, tol=None, copy_X=True, return_path=Fals
                 check_finite=False,
             )
             v = nrm2(L[n_active, :n_active]) ** 2
-            Lkk = linalg.norm(X[:, lam]) ** 2 - v
+            Lkk = linalg.norm(X[:, lam], check_finite=False) ** 2 - v
```

Comment:
This looks okay right now, because `orthogonal_mp` is the only function that calls `_cholesky_omp` and `orthogonal_mp` validates `X`. `_cholesky_omp` itself does not actually check if `X` is finite.

If in the future another function uses `_cholesky_omp` and does not validate `X`, setting `check_finite=False` here may lead to problems (crashes, non-termination)